sr_speech_control/src/sr_speech_control/speech_control.py

```diff
@@ -15,66 +15,46 @@
+        self.microphone = sr.Microphone()
+        for i, microphone_name in enumerate(sr.Microphone.list_microphone_names()):
+            if "pulse" in microphone_name:
+                self.microphone = sr.Microphone(device_index=i)
+                rospy.loginfo("Using microphone: {}".format(microphone_name))
         self.trigger_word = trigger_word
-        self.command_words = command_words
         self.recognizer = sr.Recognizer()
-        self.microphone = sr.Microphone()
         self.command_publisher = rospy.Publisher(command_topic, String, queue_size=1)
-        self.command_to_be_executed = None
-
-        self._init_recognizer(non_speaking_duration, pause_threshold)
         self._stop_listening = self.recognizer.listen_in_background(self.microphone, self._recognizer_callback)
 
-    def _init_recognizer(self, non_speaking_duration, pause_threshold):
-        with self.microphone as source:
-            self.recognizer.adjust_for_ambient_noise(source)
-        self.recognizer.non_speaking_duration = non_speaking_duration
-        self.recognizer.pause_threshold = pause_threshold
-
     def _recognizer_callback(self, recognizer, audio):
         try:
             result = recognizer.recognize_google(audio)
+            rospy.loginfo("Recognition result: {}".format(result))
         except sr.UnknownValueError:
             return
         except sr.RequestError as e:
             rospy.logwarn("Could not request results from Google Speech Recognition service: {}".format(e))
             return
 
-        result = [str(x).lower() for x in result.split(' ')]
-        if self._filter_word(result[0], self.trigger_word) == self.trigger_word:
-            command = self._filter_word(result[1], self.command_words)
-            if command in self.command_words:
-                self.command_to_be_executed = command
-
-    def _filter_word(self, word, dictionary, offset=0.5):
-        result = get_close_matches(word, dictionary, 1, offset)
-        if not result:
-            return word
-        return result[0]
+        result = result.lower()
+        if result.startswith(self.trigger_word) and len(result) > len(self.trigger_word) + 1:
+            self.command_publisher.publish(result[len(self.trigger_word) + 1:])
 
     def run(self):
         rospy.loginfo("Started speech control. Trigger word: {}".format(self.trigger_word))
         while not rospy.is_shutdown():
-                if self.command_to_be_executed:
-                    rospy.loginfo("Executing: {}.".format(self.command_to_be_executed))
-                    self.command_publisher.publish(self.command_to_be_executed)
-                    self.command_to_be_executed = None
+            rospy.sleep(1)
         self._stop_listening(wait_for_stop=False)
 
 
 if __name__ == "__main__":
-    rospy.init_node('example_speech_control', anonymous=True)
+    rospy.init_node('sr_speech_control', anonymous=True)
 
     trigger_word = "shadow"
```

Comment:
Instead of hardcoding it, we should make this a param. If we make this just a node, which I think is a good idea, we should make params here (keep "shadow" as default if desired) and run this as roslaunch, so whenever someone wants to change the default, they just pass it in the roslaunch.

Reply:
Why would someone want to change node name?

Reply:
I meant the trigger word :)

Reply:
Sorry my bad. `trigger-word` was made a command line parameter in c8fa21599e6aa78d666f81e18ceb09df8db78431.